lnwallet/chancloser/aux_closer.go

```diff
@@ -0,0 +1,52 @@
+package chancloser
+
+import (
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/fn"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/tlv"
+)
+
+// AuxCloseDesc...
+type AuxCloseDesc struct {
+	// ChanPoint...
+	ChanPoint wire.OutPoint
+
+	// Initiator...
+	Initiator bool
+
+	// CommitBlob...
+	CommitBlob fn.Option[tlv.Blob]
+
+	// FundingBlob...
+	FundingBlob fn.Option[tlv.Blob]
+}
+
+// AuxChanCloser...
+type AuxChanCloser interface {
+	// AuxCloseOpts returns the set of custom options that should be used
+	// to construct the co-op close transaction.
+	AuxCloseOpts(desc AuxCloseDesc, localShutdownBlob,
```

Comment:
We always need to have an error return type for any external hook, so we can signal the hook implementation not being ready or shutting down (instead of just blindly returning an empty option).